rust/agama-cli/src/config.rs

```diff
@@ -148,6 +157,33 @@ pub async fn run(
     }
 }
 
+/// Runs commands without remote connection to the Agama server
+pub fn run_local(subcommand: ConfigCommands, opts: GlobalOpts) -> anyhow::Result<()> {
+    match subcommand {
+        ConfigCommands::Validate { url_or_path } => validate_json(url_or_path, opts.insecure),
+        _ => {
+            eprintln!("This subcommand doesn't support --local option");
+            Ok(())
+        }
+    }
+}
+
+/// Validates a JSON profile with locally available tools only
+fn validate_json(url_or_path: CliInput, insecure: bool) -> anyhow::Result<()> {
```

Comment:
I do not like the name as there are already validate...maybe I would use something like `validate_locally` ? that emphasize, that it is validation for local only purpose.